dvc/testing/test_api.py

```diff
@@ -38,7 +38,7 @@ def test_open(self, tmp_dir, dvc, remote):
     )
     def test_filesystem(
         self,
-        M,
+        M,  # from `pytest-test-utils`
```

Comment:
```suggestion
        M,
```

Please don't add this. You'd need to do the same for every fixtures, where does `tmp_dir` come from, for example?

Reply:
One can search for `tmp_dir`, searching for `M` is something else. I can get rid of that comment, but I would consider renaming the fixture to something meaningful (and searchable)

Reply:
How can you search for `tmp_dir` in `dvc-s3` and find it? I’m happy to rename it to `match` (there’s a `Matcher` alias already) but I don’t think that would solve discoverability issue.

I do agree that the naming is odd but that’s a tradeoff that we chose and once you are familiar with it, it’s no longer an issue.
Again, I can’t solve the discoverability issue in pytest-fixture.

Reply:
The best way would be to add typehints, but I don’t want to force them where we don’t use them in the tests.

Reply:
Well, one can easily get the source of the fixture by looking at `pytest --fixtures`:
> ...
> ------------------------------------------------ fixtures defined from pytest_test_utils.pytest_plugin ------------------------------------------
> tmp_dir_factory [session scope] -- .venv/lib/python3.10/site-packages/pytest_test_utils/pytest_plugin.py:13
>    no docstring available
>
> tmp_dir -- .venv/lib/python3.10/site-packages/pytest_test_utils/pytest_plugin.py:18
>   no docstring available
>
> matcher -- .venv/lib/python3.10/site-packages/pytest_test_utils/pytest_plugin.py:25
>    no docstring available
>
> M -- .venv/lib/python3.10/site-packages/pytest_test_utils/pytest_plugin.py:30
 >    no docstring available
 > ...
 
Although the name `M` still does not help. I understand the "once you use it you know what it is" point, but I'm not sure what the tradeoff is here (not having to type a few extra characters?).
 
 `match` is a keyword in 3.10, so I'd avoid it, why not just use `matcher`, since it's the same thing?

Reply:
> Well, one can easily get the source of the fixture by looking at pytest --fixtures:

That works for whatever fixture name that you have. But you can never know what a fixture is from the comfort of your IDEs/text-editors, that's the magic of pytest that I _cannot_ solve. So this comment does not help.
If you feel like we need to solve the "discoverability" issue, add/propose type-hints in the tests. Again, remove this comment, please, it's not helping.

> Although the name M still does not help. I understand the "once you use it you know what it is" point, but I'm not sure what the tradeoff is here (not having to type a few extra characters?).

This `Matcher` is meant to be used on comparing unstructured, nested data for things that you care about.
It is used all over `data_status()` tests:

https://github.com/iterative/dvc/blob/197f522175eca5fd33aed47e1f89730ab8bf12e7/tests/func/test_data_status.py#L254-L260

This is meant to be used multiple times, even in a single assertion. So it saves a lot of characters, is easier to look at and read, etc. After you are familiar with it, `Matcher` just becomes noise and the `Matcher` vs `M` becomes a minor concern.

These types of naming, even though it's rare, you can find a lot of examples in the Python world:
- [`django.db.models.F`](https://docs.djangoproject.com/en/4.1/ref/models/expressions/#django.db.models.F)
- [`glom.M`](https://glom.readthedocs.io/en/latest/matching.html#glom.M) and [`glom.T`](https://glom.readthedocs.io/en/latest/api.html#glom.T)

I know the naming goes against standard naming conventions, but sometimes it's okay to break one, especially for tests. Tbh I don't like it either (see https://github.com/iterative/studio/pull/2886), but I traded it off for better test readability.

I am only trying to provide the context here. If you are not convinced or if the majority disagrees, feel free to create a PR to rename it to `matcher`. But this comment is not the solution.